esx_service/vmdk_ops.py

```diff
@@ -837,12 +837,33 @@ def parse_vol_name(full_vol_name):
             raise ValidationError("Volume name contains illegal characters: {0}".format(c))
 
     # Check if the datastore name is too long
-    if ds_name and len(ds_name) > MAX_DS_NAME_LEN:
-        raise ValidationError("Datastore name is too long (max len is {0})".format(MAX_DS_NAME_LEN))
+    if ds_name:
+        if len(ds_name) > MAX_DS_NAME_LEN:
+            raise ValidationError("Datastore name is too long (max len is {0})".format(MAX_DS_NAME_LEN))
+
+        matching_datastores = find_matching_datastores(ds_name)
+
+        # Return error if more than one datastores found
+        if len(matching_datastores) > 1:
+            raise ValidationError("Found multiple datastores with same name (ignoring case difference): {0}".format(matching_datastores))
+
+        # Found exactly one match
+        if len(matching_datastores) == 1:
+            ds_name = matching_datastores[0]
```

Comment:
my 2 cents ... 

datastore name is case insensitive from vc/esx point of view. You are able to keep 2 datastore  (datastore 1 and datastore2) name as `localDatastore` & `LocalDatastore` any point of time no restriction to have such names for datastore. 

/CC @shaominchen @msterin